### PR TITLE
AVM Plugin Update

### PR DESCRIPTION
**Pull request by sisamiwe**
- Update Docu
 - Update WebIF
 - Connect to Smarthome Devices via AHA-Interface (tested mit Fritz 440, Comet Dect) (therefore Plugin AVM smarthome can be depreciated)
 - implement call deflections (tested mit Fritzbox 7530)
 - Rename Parameter "index" to "avm_tam_index"
 - Code Cleanup (add new functions to minimize code repetitions)
 - Improve error handling

### CODE REVIEW

**Comment by psilo909** [comment]
Also starting a review now. hopefully until weekend. at least i now integrated the PR and will observe with my standard features

**Comment by sisamiwe** [comment]
> And the first error:

This could be solved by adding `if xml is not None:` before line 1965 as it is an the other xml parsings.

**Comment by psilo909** [comment]
> > And the first error:
> 
> This could be solved by adding `if xml is not None:` before line 1965 as it is an the other xml parsings.

I will try tomorrow and get back to you. Then you can add it to PR.

**Comment by sisamiwe** [comment]
> and another different one:

This has been the implementation for Fritz 440, done with @bmxp. I can extend this for other devices, if I know their repsonse.

**Comment by sisamiwe** [comment]
> and some 404. What do i have to change for AHA stuff?

Do you have any avm_data_type defined for these devices? In plugin.yaml I noted down, which avm_data_type is using which interface.

**Comment by psilo909** [comment]
> wlan_repeater_2400_garage

no, the items below them are just our mobile phones (for presence detection)

**Comment by sisamiwe** [comment]
@psilo909 

Could me move that commication to gitter?

**Comment by sisamiwe** [comment]
> Example: wlan_repeater_2400_garage

Could you check at the "[AVM Smarthome Devices]" Tab of the WebIF, which smarthome devices and their functions were detected?

For my hkr it looks like: `{'device_id': '19', 'fw_version': '03.54', 'product_name': 'Comet DECT', 'manufacturer': 'AVM', 'functions': ['thermostat', 'temperature_sensor'], 'batterylow': False, 'battery_level': 70, 'connected': True, 'tx_busy': False, 'device_name': 'OG Bad', 'current_temperature': 21.5, 'target_temperature': 20.0, 'temperature_comfort': 21.0, 'temperature_reduced': 16.0, 'battery_low': False, 'window_open': False, 'summer_active': False, 'holiday_active': False, 'boost_active': False, 'lock': False, 'device_lock': False, 'errorcode': 0, 'windowopenactiveendtime': 0, 'boostactiveendtime': 0, 'temperature_offset': 0.0}`

**Comment by aschwith** [comment]
> > And the first error:
> 
> This could be solved by adding `if xml is not None:` before line 1965 as it is an the other xml parsings.

Korrekt. I also had to add an equivalent: 
"if not xml: 
            return
"
inside the _update_host(self, item) function. @sisamiwe : Could you add both to the pull request.

**Comment by psilo909** [comment]
hab mal einiges in den smarthome channel geschrieben

**Comment by psilo909** [comment]
fakt ist, bei mir taucht im webif nur 1 device auf anstatt 7 wie im router zu sehen

File: avm/__init__.py [hunk 1]
```diff
@@ -2421,12 +2425,12 @@ def _update_aha_devices(self):
 
     def _update_smarthome_items(self):
         """Update smarthome items using dict '_smarthome_devices'"""
-        for item in self._fritz_device._smarthome_items:
+        for item in self._fritz_device.get_smarthome_devices():
```
Comment (psilo909) [inline avm/__init__.py:2428]: probably this is causing my 
```
2022-02-13  09:02:05 ERROR    plugins.avm_fritzbox_7590.update Method plugins.avm_fritzbox_7590.update exception: 'str' object has no attribute 'conf'
> Traceback (most recent call last):
>   File "/python/smarthome/lib/scheduler.py", line 657, in _task
>     obj()
>   File "/python/smarthome/plugins/avm/__init__.py", line 783, in _update_loop
>     self._update_aha_devices()
>   File "/python/smarthome/plugins/avm/__init__.py", line 2424, in _update_aha_devices
>     self._update_smarthome_items()
>   File "/python/smarthome/plugins/avm/__init__.py", line 2430, in _update_smarthome_items
>     ainDevice = self._get_item_ain(item)
>   File "/python/smarthome/plugins/avm/__init__.py", line 2542, in _get_item_ain
>     if self.has_iattr(item.conf, 'ain'):
```
now

Comment (psilo909) [reply avm/__init__.py:2428]: it looks like this is a dict, therefore here we just get AIN and not items:
self._fritz_device.get_smarthome_devices()[ain]

**Pull request merged**